scipy/_lib/tests/test_public_api.py

```diff
@@ -212,16 +214,33 @@ def is_unexpected(name):
 ]
 
 
+# XXX: this test does more than it says on the tin - in using `pkgutil.walk_packages`,
+# it will raise if it encounters any exceptions which are not of type `ImportError` or
+# handled by `ignore_errors` while attempting to import each discovered package.
+# For now, `ignore_errors` only ignores what is necessary, but this could be expanded -
+# for example, to all errors from private modules or git subpackages - if desired.
```

Comment:
This was not quite correct - to maintain the default behaviour of ignoring `ImportError`s we would have to handle that in `ignore_errors` (but looks like we don't need to right now).
```suggestion
# XXX: this test does more than it says on the tin - in using `pkgutil.walk_packages`,
# it will raise if it encounters any exceptions which are not handled by `ignore_errors`
# while attempting to import each discovered package.
# For now, `ignore_errors` only ignores what is necessary, but this could be expanded -
# for example, to all errors from private modules or git subpackages - if desired.
```